src/etc/inc/interfaces.inc

```diff
@@ -6000,12 +6005,13 @@ function get_interface_vendor_mac($interface) {
 	global $g;
 
 	$mac = "";
-	$mac_file = "{$g['cf_conf_path']}/vendor_mac_{$interface}";
+	$mac_file = "{$g['vardb_path']}/vendor_mac";
 	if (file_exists($mac_file)) {
-		$mac = file_get_contents($mac_file);
-		if (!is_macaddr($mac)) {
-			$mac = "";
-		}
+		$vendor_mac_contents = file_get_contents($mac_file);
+		$mac_pattern = '(?:[[:xdigit:]]{2}:){5}[[:xdigit:]]{2}';
+		$pattern = '/^'.$interface.': ('.$mac_pattern.')$/m';
```

Comment:
It could be simpler using file() instead of file_get_contents() like:

```php
$vendor_mac_lines = file("{$g['vardb_path']}/vendor_mac",
    FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
````